Reduced unnecessary UI updates

Mailaender opened the pull request:
Fixes https://github.com/eclipse/chemclipse/issues/540

Mailaender committed:
Only update targets when there are actual changes.

Signed-off-by: Matthias Mailänder <matthias.mailaender@lablicate.com>

Mailaender committed:
Only update when the chromatogram selection differs.

Signed-off-by: Matthias Mailänder <matthias.mailaender@lablicate.com>

eselmeister commented:
@Mailaender The change in the chromatgram selection looks fine. The only issue I've found is, that the scan targets part and chart/table is not updated when switching to a reference chromatogram.

![Bildschirmfoto von 2021-03-24 16-48-34](https://user-images.githubusercontent.com/686671/112343801-c36acd80-8cc3-11eb-8f06-6d18de430786.png)

As this change leads to a better usability of the chromatogram popup menu, it's IHMO acceptable.

But I would strongly recommend to skip the cache handling in the TargetsPart, as the TargetSupplier could be the same but some target labels might have been modified or e.g. the mol weight was calculated for each entry via the chromatogram menu. It might be confusing if the changes in the targets entries are not displayed.

![Bildschirmfoto von 2021-03-24 17-14-26](https://user-images.githubusercontent.com/686671/112344600-8c48ec00-8cc4-11eb-8dbc-5fae44700ef8.png)

It should be updated after running a filter to:

![Bildschirmfoto von 2021-03-24 17-14-34](https://user-images.githubusercontent.com/686671/112344648-979c1780-8cc4-11eb-87f5-4c6330610409.png)

Mailaender committed:
Upddate scan chart/table when chromatogram (reference) changes.

Signed-off-by: Matthias Mailänder <matthias.mailaender@lablicate.com>

Mailaender committed:
Don't spam updates when there are no targets!

See propagateTarget which will already notify the UI.

Signed-off-by: Matthias Mailänder <matthias.mailaender@lablicate.com>

Mailaender commented:
I think I finally identified and removed the real culprit. https://github.com/eclipse/chemclipse/pull/543/commits/88ee2c5c27842f13fe27ab2b27404926e2813971 A lot of the UI updates indeed seem to rely on this spamming events even when the part is empty. I added updates for scan table/chart similar to the peak list which already behaves properly. I also don't cache the individual targets anymore only the chromatogram. I haven't tested the mol weight calculation as my current setup in Eclipse is either ChemClipse or OpenChrom not mixed together.
